### PR TITLE
Judicial-system/Strict TS setting pt2

### PR DESCRIPTION
**Pull request by oddsson**
# Prosecutor flow now uses strict TS

https://app.asana.com/0/1197628843274291/1199143603683037

## What

See ☝️

## Why

It's a requirement from island-is. Also it's generally good practice.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepTwo/StepTwo.tsx [hunk 1]
```diff
@@ -522,40 +537,49 @@ export const StepTwo: React.FC = () => {
                             const cRestrictions =
                               copyOfState.custodyRestrictions
 
-                            restrictions.splice(
-                              restrictions.indexOf(
-                                target.value as CaseCustodyRestrictions,
-                              ),
-                              1,
-                            )
+                            if (restrictions) {
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepTwo/StepTwo.tsx:540]: Why guard against this here, but not above?

File: apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepTwo/StepTwo.tsx [hunk 2]
```diff
@@ -522,40 +537,49 @@ export const StepTwo: React.FC = () => {
...
+
                           // TODO: COMBINE IN A SINGLE API CALL
-                          await updateCase(
-                            workingCase.id,
-                            parseArray(
-                              'custodyRestrictions',
-                              copyOfState.custodyRestrictions,
-                            ),
-                          )
+                          if (copyOfState.custodyRestrictions) {
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepTwo/StepTwo.tsx:574]: Consider copying the value in the judge flow. By copying this way, any values set by the judge will be overwritten here. Or am I missing something?

**Review by gudjong — COMMENTED** [review]
Have a couple of comments.

File: apps/judicial-system/web/src/routes/Judge/Ruling/StepOne/RulingStepOne.tsx [hunk 3]
```diff
@@ -42,11 +42,14 @@ import {
...
 } from '@island.is/judicial-system-web/src/types'
 
+interface CaseData {
+  case: Case
+}
+
 export const RulingStepOne: React.FC = () => {
   const custodyEndTimeRef = useRef<HTMLInputElement>()
-  const [workingCase, setWorkingCase] = useState<Case>()
+  const [workingCase, setWorkingCase] = useState<Case>(null)
```
Comment (addi) [inline apps/judicial-system/web/src/routes/Judge/Ruling/StepOne/RulingStepOne.tsx:51]: Vorum við ekki búin að leysa eitthvað svona?

File: apps/judicial-system/web/src/routes/Judge/Ruling/StepOne/RulingStepOne.tsx [hunk 4]
```diff
@@ -112,15 +113,26 @@ export const RulingStepOne: React.FC = () => {
...
 
   useEffect(() => {
-    const getCurrentCase = async () => {
-      setIsLoading(true)
-      setWorkingCase(resCase)
-      setIsLoading(false)
-    }
-    if (id && !workingCase && resCase) {
-      getCurrentCase()
+    if (id && workingCase === null && data) {
```
Comment (addi) [inline apps/judicial-system/web/src/routes/Judge/Ruling/StepOne/RulingStepOne.tsx:116]: Þarft ekki að hafa id checkið þarna. Það er ekkert working case án id, id er hvort sem er ekki notað í effectinu.

File: apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepOne/StepOne.tsx [hunk 5]
```diff
@@ -109,7 +109,7 @@ export const CreateCaseMutation = gql`
 
 export const StepOne: React.FC = () => {
   const history = useHistory()
-  const [workingCase, setWorkingCase] = useState<Case | null>(null)
+  const [workingCase, setWorkingCase] = useState<Case>()
```
Comment (addi) [inline apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepOne/StepOne.tsx:112]: Hérna erum við td ekki með null í default value. Sjá comment á RulingStepOne.tsx

**Review by addi — CHANGES_REQUESTED** [review]
Vantar smá púss.

**Review by addi — APPROVED** [review]

**Pull request merged**